File: Java/Sensores/01_Ud5_ListarSensores/app/src/main/java/com/pms/a01_ud5_listarsensores/MainActivity.java
package com.pms.a01_ud5_listarsensores;

import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import java.util.List;

public class MainActivity extends AppCompatActivity {

  private TextView etiqueta;
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    etiqueta = (TextView) findViewById(R.id.listado);
        /* Usar getSystemService() con la constante SENSOR_SERVICE y obtener
        el objeto sensorManager  que accederá a la lista de servicios del sistema.*/
    SensorManager sensorManager = (SensorManager)
        getSystemService(SENSOR_SERVICE);
    // obtener la lista de todos los sensores   (TYPE_ALL)
    List<Sensor> listaSensores = sensorManager.getSensorList(Sensor.TYPE_ALL);
    // recorrer todos los sensores
    for(Sensor sensor: listaSensores)
      // añadir el sensor al label o etiqueta
      etiqueta.append("\n-" + sensor.getName() + ", fabricante: " +sensor.getVendor());

  }
}

File: Kotlin/Sensores/01_Ud5_ListarSensores/app/src/main/java/com/pms/a01_ud5_listarsensores/MainActivity.kt
package com.pms.a01_ud5_listarsensores

import android.hardware.Sensor
import android.hardware.SensorManager
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    private var etiqueta: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        etiqueta = findViewById<View>(R.id.listado) as TextView
        /* Usar getSystemService() con la constante SENSOR_SERVICE y obtener
        el objeto sensorManager  que accederá a la lista de servicios del sistema.*/
        val sensorManager = getSystemService(SENSOR_SERVICE) as SensorManager
        // obtener la lista de todos los sensores   (TYPE_ALL)
        val listaSensores = sensorManager.getSensorList(Sensor.TYPE_ALL)
        // recorrer todos los sensores
        for (sensor in listaSensores)  // añadir el sensor al label o etiqueta
            etiqueta!!.append("""
    
    -${sensor.name}, fabricante: ${sensor.vendor}
    """.trimIndent())
    }
}